python/ray/_private/worker.py

```diff
@@ -846,6 +851,62 @@ def get_gpu_ids():
     Returns:
         A list of GPU IDs.
     """
+    assigned_ids = _get_all_resource_ids("GPU", r"^GPU_group_[0-9A-Za-z]+$")
+    # If the user had already set CUDA_VISIBLE_DEVICES, then respect that (in
+    # the sense that only GPU IDs that appear in CUDA_VISIBLE_DEVICES should be
+    # returned).
+    if global_worker.original_gpu_ids is not None:
+        assigned_ids = [
+            global_worker.original_gpu_ids[gpu_id] for gpu_id in assigned_ids
+        ]
+        # Give all GPUs in local_mode.
+        if global_worker.mode == LOCAL_MODE:
+            max_gpus = global_worker.node.get_resource_spec().num_gpus
+            assigned_ids = global_worker.original_gpu_ids[:max_gpus]
+
+    return assigned_ids
+
+
+@PublicAPI
+@client_mode_hook
+def get_neuron_core_ids():
+    """Get the IDs of the NeuronCores that are available to the worker.
+
+    If the NEURON_RT_VISIBLE_CORES environment variable was set when the worker
+    started up, then the IDs returned by this method will be a subset of the
+    IDs in NEURON_RT_VISIBLE_CORES. If not, the IDs will fall in the range
+    [0, NEURON_CORES - 1], where NEURON_CORES is the number of neuron_cores
+    that the node has.
+
+    Returns:
+        A list of NEURON_CORE IDs.
+    """
+    neuron_regex = (
+        f"^{ray._private.utils.get_neuron_core_constraint_name()}_group_[0-9A-Za-z]+$"
+    )
+    assigned_ids = _get_all_resource_ids(ray_constants.NUM_NEURON_CORES, neuron_regex)
+    # If the user had already set , then respect that
+    # (in the sense that only neuron-core IDs that appear in
+    # NEURON_RT_VISIBLE_CORES should be returned).
+    if global_worker.original_aws_neuron_core_ids is not None:
+        assigned_ids = [
+            global_worker.original_aws_neuron_core_ids[neuron_core_id]
+            for neuron_core_id in assigned_ids
+        ]
+        # Give all neuron_cores in local_mode.
+        if global_worker.mode == LOCAL_MODE:
+            max_neuron_cores = global_worker.node.get_resource_spec().resources.get(
+                ray_constants.NUM_NEURON_CORES, None
+            )
+            if max_neuron_cores:
+                assigned_ids = global_worker.original_aws_neuron_core_ids[
+                    :max_neuron_cores
+                ]
+
+    return assigned_ids
+
+
+def _get_all_resource_ids(resource_name: str, resource_regex: str):
```

Comment:
I see some friction here. Currently I'm unable to get multiple resource/ids for custom resources.
For example,
```
@ray.remote(num_gpus=2)
class Actor0
...

_returns_
{'GPU': [(0, 1.0), (1, 1.0)], 'CPU': [(0, 1.0)]}
```
whereas

```
@ray.remote(resources={"num_neuron_cores": 2})
class Actor0
...

_returns_
{'CPU': [(0, 1.0)], 'num_neuron_cores': [(0, 2.0)]}
```
which would consistently reconfigure core_0 for Neuron

1. Do I have modify any low-level code to reflect the changes? 
2. Regarding `accelerator_type`,  what's motivation of `0.001`?  `{'num_neuron_cores': [(0, 2.0)], 'CPU': [(0, 1.0)], 'accelerator_type:aws-neuron-core': [(0, 0.001)]}`

Reply:
Reg (1), able to find an alternative by adding `num_neuron_cores` to `custom_unit_instance_resources`

Reply:
> Regarding accelerator_type, what's motivation of 0.001? {'num_neuron_cores': [(0, 2.0)], 'CPU': [(0, 1.0)], 'accelerator_type:aws-neuron-core': [(0, 0.001)]}

This is a hack for Nvidia GPUs to indicate not only need GPU, but also need specify accelerator_type because we don't have label yet.

Reply:
(so in aws_accelerator's case, we don't need to worry about it)

Reply:
Thanks for the confirmation